WordPress/wordpress-develop
# Fix wrong file size when scaling images.

### What
This PR fixes the wrong file size displayed in the media library when a user resizes an image. At the same time, if the user decides to revert the changes made to the image, this PR makes possible to revert to the original file size.

### Why
This PR solves the issue described in the following Trac ticket: https://core.trac.wordpress.org/ticket/59684

### How
* In the `wp_save_image` function I collect the array returned by `wp_save_image_file`, which contains the file size of the resized image,  and add it to the attachment metadata array as `$meta['filesize']`
  * This allows to show the correct file size when the image is scaled
* Before doing this I save the original file size in the `$backup_sizes` array, along with the image filename and dimensions
  * This allows to retrieve the original file size in case the user reverts the changes made to the image size
* In the `wp_restore_image` function I then read back from `$backup_sizes` the original file size and restore it when the user reverts back to the original size

### Test instructions
<details>
  <summary>Test the scale image case</summary>

* Go to `Media` -> `Add New Media File` and upload an image
  * take note of the image size
  <img width="686" alt="Screenshot 2024-04-18 at 23 12 22" src="https://github.com/WordPress/wordpress-develop/assets/68744851/629f84a2-2301-4ad1-bc94-8c630c2c58d5">
* Select the image you just uploaded and click on `Edit Image`
  * click on `Scale` and specify the new width and height
* Close the modal and select the image you just resized  
  * verify the file size is decreased
  <img width="739" alt="Screenshot 2024-04-18 at 23 16 48" src="https://github.com/WordPress/wordpress-develop/assets/68744851/2e1952e7-aaa5-452f-a59e-f29c3b6f3118">

</details>

<details>
  <summary>Test the restore original image case</summary>

* Select the image you just resized and click on `Edit Image`
  * click on `RESTORE ORIGINAL IMAGE` -> `Restore Image`
* Close the modal and select the image you just restored  
  * verify the file size is back to its original value
  
</details>

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
